source/extensions/quic_listeners/quiche/platform/quiche_unordered_containers_impl.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "absl/container/node_hash_map.h"
+
+namespace quiche {
+
+// The default hasher used by hash tables.
+template <typename Key> using QuicheDefaultHasherImpl = absl::Hash<Key>;
+
+// Similar to std::unordered_map, but with better performance and memory usage.
+template <typename Key, typename Value, typename Hash>
+using QuicheUnorderedMapImpl = absl::node_hash_map<Key, Value, Hash>;
```

Comment:
Is there a particular reason to use absl::node_hash_map instead of absl::flat_hash_map?

Reply:
SGTM. In the long term we should probably rename it to QuicheNodeHashMap and add a QuicheFlatHashMap, or directly use the absl ones if that's allowed.